boofuzz/primitives/fuzz_list.py

```diff
@@ -0,0 +1,125 @@
+import random
+import glob
+
+from .base_primitive import BasePrimitive
+
+
+class FuzzList(BasePrimitive):
+    # store fuzz_library as a class variable to avoid copying the ~70MB structure across each instantiated primitive.
+    _fuzz_library = []
+
+    def __init__(self, value, size=-1, padding="\x00", encoding="ascii", fuzzable=True, max_len=0, name=None, filename=None):
+        """
+        Primitive that cycles through a list of "bad" values from a file. The primitive take filename and open the file to read
+        the values to use in fuzzing process.
+
+        @type  value:    str
+        @param value:    Default string value
+        @type  size:     int
+        @param size:     (Optional, def=-1) Static size of this field, leave -1 for dynamic.
+        @type  padding:  chr
+        @param padding:  (Optional, def="\\x00") Value to use as padding to fill static field size.
+        @type  encoding: str
+        @param encoding: (Optional, def="ascii") String encoding, ex: utf_16_le for Microsoft Unicode.
+        @type  fuzzable: bool
+        @param fuzzable: (Optional, def=True) Enable/disable fuzzing of this primitive
+        @type  max_len:  int
+        @param max_len:  (Optional, def=0) Maximum string length
+        @type  name:     str
+        @param name:     (Optional, def=None) Specifying a name gives you direct access to a primitive
+        @type  filename: str
+        @param filename: Filename pattern to load all fuzz value
+        """
+
+        super(FuzzList, self).__init__()
+
+        self._value = self._original_value = value
+        self.size = size
+        self.padding = padding
+        self.encoding = encoding
+        self._fuzzable = fuzzable
+        self._name = name
+        self._filename = filename
+        self.this_library = []
+        list_of_files = glob.glob(self._filename)
+        for fname in list_of_files:
+            _file_handle = open(fname,"r")
+            self.this_library.append(_file_handle.readlines())
```

Comment:
I think you want `.extend` instead of `.append`. `readlines()` returns a list, so using `append` will make `self.this_library` a list of lists. Example:
```python
>>> a = [1,2,3]
>>> a.extend([4,5])
>>> a
[1, 2, 3, 4, 5]
>>> a.append([6,7])
>>> a
[1, 2, 3, 4, 5, [6, 7]]
```